portfolio/src/main/webapp/app.js

```diff
@@ -64,10 +64,25 @@ function searchBooks(book, url)
         var publishedDate = info.publishedDate;
         var publisher = info.publisher;
         var maturityRating = info.maturityRating;
+
+        var fieldNameArray = ["title", "genre", "categories", "author", "language", "description", "infoLink", "pageCount", "publishedDate", "publisher", "maturityRating"];
```

Comment:
Kinda the same thing here as my comment in the servlet. It's particularly fragile maintaining these as two separate arrays. If either of them were to get out of sync, we'd have some issues in the loops below.

const BOOK_FIELDS = {
    TITLE: 'title',
    ...
}